include/nvexec/stream/scan.cuh

```diff
@@ -0,0 +1,172 @@
+            Fun,
+            receiver_t<SenderId, ReceiverId, InitT, Fun>> {
+        using base = __algo_range_init_fun::
+          receiver_t<SenderId, ReceiverId, InitT, Fun, receiver_t<SenderId, ReceiverId, InitT, Fun>>;
+
+        template <class Range>
+        using result_t = typename __algo_range_init_fun::binary_invoke_result_t<Range, InitT, Fun>;
+
+        template <class Range>
+        static void set_value_impl(base::__t&& self, Range&& range) noexcept {
+          cudaError_t status{cudaSuccess};
+          cudaStream_t stream = self.op_state_.get_stream();
+
+          // `range` is produced asynchronously, so we need to wait for it to be ready
+          if (status = STDEXEC_DBG_ERR(cudaStreamSynchronize(stream)); status != cudaSuccess) {
+            self.op_state_.propagate_completion_signal(stdexec::set_error, std::move(status));
+            return;
+          }
+
+          using value_t = result_t<Range>;
+          value_t* d_out = static_cast<value_t*>(self.op_state_.temp_storage_);
+
+          void* d_temp_storage{};
+          std::size_t temp_storage_size{};
+
+          auto first = begin(range);
+          auto last = end(range);
+
+          std::size_t num_items = std::distance(first, last);
+
+          if (status = STDEXEC_DBG_ERR(cub::DeviceScan::ExclusiveScan(
+                d_temp_storage,
+                temp_storage_size,
+                first,
+                d_out,
+                self.fun_,
+                self.init_,
+                num_items,
+                stream));
+              status != cudaSuccess) {
+            self.op_state_.propagate_completion_signal(stdexec::set_error, std::move(status));
+            return;
+          }
+
+          if (status = STDEXEC_DBG_ERR( //
+                cudaMallocAsync(&d_temp_storage, temp_storage_size, stream));
+              status != cudaSuccess) {
+            self.op_state_.propagate_completion_signal(stdexec::set_error, std::move(status));
+            return;
+          }
+
+          if (status = STDEXEC_DBG_ERR(cub::DeviceScan::ExclusiveScan(
+                d_temp_storage,
+                temp_storage_size,
+                first,
+                d_out,
+                self.fun_,
+                self.init_,
+                num_items,
+                stream));
+              status != cudaSuccess) {
+            self.op_state_.propagate_completion_signal(stdexec::set_error, std::move(status));
+            return;
+          }
+
+          status = STDEXEC_DBG_ERR(cudaFreeAsync(d_temp_storage, stream));
+          self.op_state_.defer_temp_storage_destruction(d_out);
+          if (status == cudaSuccess) {
+            int* host_out = new int[10];
+            cudaMemcpy(host_out, d_out, 10 * sizeof(int), cudaMemcpyDeviceToHost);
+            for (size_t i = 0; i < 10; ++i) {
+              std::cout << host_out[i] << std::endl;
+            }
+            self.op_state_.propagate_completion_signal(stdexec::set_value, d_out);
```

Comment:
If you change `d_out` to `*d_out` you can confirm that the data is scanning properly. But it won't compile I imagine because of the comcpletion signatures being wrong.